pkg/csi/service/common/common_controller_helper.go

```diff
@@ -317,34 +318,35 @@ func IsVolumeSnapshotReady(ctx context.Context, client snapshotterClientSet.Inte
 	isReadyToUse := false
 	var svs *snap.VolumeSnapshot
 
-	waitErr := wait.PollImmediate(5*time.Second, timeout, func() (done bool, err error) {
-		svs, err = client.SnapshotV1().VolumeSnapshots(namespace).
-			Get(ctx, supervisorVolumeSnapshotName, metav1.GetOptions{})
-		if err != nil {
-			msg := fmt.Sprintf("unable to fetch volumesnapshot %q/%q "+
-				"from supervisor cluster with err: %+v",
-				namespace, supervisorVolumeSnapshotName, err)
-			log.Warnf(msg)
-			return false, logger.LogNewErrorf(log, msg)
-		}
-		if svs == nil || svs.Status == nil || svs.Status.ReadyToUse == nil {
-			log.Infof("Waiting up to %d seconds for VolumeSnapshot %v in namespace %s to be ReadyToUse, %+vs "+
-				"since the start time", timeoutSeconds, supervisorVolumeSnapshotName, namespace,
-				time.Since(startTime).Seconds())
+	waitErr := wait.PollUntilContextTimeout(ctx, 5*time.Second, timeout, true,
```

Comment:
why do we need to supply timeout as arg to `PollUntilContextTimeout`? isn't it getting timeout from context?

Reply:
`timeout` is a compulsory argument to `PollUntilContextTimeout`. The method uses the timeout value to create an internal context with the supplied context used as a parent. 

```
func PollUntilContextTimeout(ctx context.Context, interval, timeout time.Duration, immediate bool, condition ConditionWithContextFunc) error {
	deadlineCtx, deadlineCancel := context.WithTimeout(ctx, timeout)
	defer deadlineCancel()
	return loopConditionUntilContext(deadlineCtx, Backoff{Duration: interval}.Timer(), immediate, false, condition)
}
```